### PR TITLE
feat(suite): Virtualized token select

### PR DESCRIPTION
**Pull request by jvaclavik**
## Related Issue

Resolve 

## Screenshots:
image

### CODE REVIEW

File: packages/product-components/src/components/SelectAssetModal/AssetItem.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                </Column>
+                <div>
+                    <Badge>{badge}</Badge>
+                </div>
+                {isFavorite ? (
+                    <Icon
+                        name="starFilled"
+                        color={theme.backgroundAlertYellowBoldAlt}
+                        onClick={handleFavoriteClick}
+                        cursorPointer
```
Comment (seibei-iguchi) [inline packages/product-components/src/components/SelectAssetModal/AssetItem.tsx]: This prop is deprecated, I think.

File: packages/product-components/src/components/SelectAssetModal/SelectAssetModal.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    VirtualizedList,
+} from '@trezor/components';
+import { mapElevationToBackgroundToken, spacings, spacingsPx, typography } from '@trezor/theme';
+import styled from 'styled-components';
+import { AssetItem } from './AssetItem';
+
+const LIST_HEIGHT = '50vh';
+const height = 60;
+
+const NoData = styled.div`
```
Comment (seibei-iguchi) [inline packages/product-components/src/components/SelectAssetModal/SelectAssetModal.tsx]: Use Column and Paragraph instead?

File: packages/product-components/src/components/SelectAssetModal/SelectAssetModal.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                                    handleClick={onSelectAssetModal}
+                                    onFavoriteClick={onFavoriteClick}
+                                />
+                            )}
+                            onScrollEnd={() => {
+                                setEnd(end + 1000);
+                            }}
+                            height={LIST_HEIGHT}
+                        />
+                        <ShadowBottom backgroundColor="backgroundSurfaceElevation0" />
```
Comment (seibei-iguchi) [inline packages/product-components/src/components/SelectAssetModal/SelectAssetModal.tsx]: Use mapElevationToBackgroundToken

**Comment by adderpositive** [comment]
- bug, when I am searching for one item (**name 0**) in an array 
  - `if (!items[itemIndex]) return null;` - on the line 135th in `VirtualizeList` will fix it (maybe some cleaner way)
- there should be input data - select array and group array (we need for our purpose to group the assets)
- remove `any`s
- optional `onFavoriteClick` -  do it like `isFavouriteActive` and logic could be in the modal, what do you think?
- add `isActive` 
- cancel search input doesn't reset the list
- active selected network should look differently and should not be in text search (can not search token in a selected network)
- some design inaccuracies (alignment of a badge, error messages, line under networks), check the design here https://www.figma.com/design/xSW1knAjH25SfOmqSdNioY/Trade?node-id=6032-33921&m=dev

File: packages/product-components/src/components/SelectAssetModal/AssetItem.tsx (hunk 1 above)
Comment (jvaclavik) [reply packages/product-components/src/components/SelectAssetModal/AssetItem.tsx]: Done

File: packages/product-components/src/components/SelectAssetModal/SelectAssetModal.tsx (hunk 2 above)
Comment (jvaclavik) [reply packages/product-components/src/components/SelectAssetModal/SelectAssetModal.tsx]: Done

File: packages/product-components/src/components/SelectAssetModal/SelectAssetModal.tsx (hunk 3 above)
Comment (jvaclavik) [reply packages/product-components/src/components/SelectAssetModal/SelectAssetModal.tsx]: Done

**Review by seibei-iguchi — APPROVED** [review]

**Comment by jvaclavik** [comment]
Thanks 🙏 

> * bug, when I am searching for one item (**name 0**) in an array
>   * `if (!items[itemIndex]) return null;` - on the line 135th in `VirtualizeList` will fix it (maybe some cleaner way)

Done

> * there should be input data - select array and group array (we need for our purpose to group the assets)

Discussed in DM

> * remove `any`s

Done

> * optional `onFavoriteClick` -  do it like `isFavouriteActive` and logic could be in the modal, what do you think?

Done

> * add `isActive`

Discussed in DM

> * cancel search input doesn't reset the list

Discussed in DM

> * active selected network should look differently and should not be in text search (can not search token in a selected network)

Discussed in DM

> * some design inaccuracies (alignment of a badge, error messages, line under networks), check the design here https://www.figma.com/design/xSW1knAjH25SfOmqSdNioY/Trade?node-id=6032-33921&m=dev

Discussed in DM

**Pull request merged**